omod-2.0/src/test/java/org/openmrs/module/webservices/rest/web/v1_0/controller/openmrs2_0/PatientAllergyController2_0Test.java

```diff
@@ -0,0 +1,341 @@
+		allergies = Context.getPatientService().getAllergies(patient);
+		Assert.assertEquals(allergies.size(), 5);
+	}
+	
+	/**
+	 *  Save Allergy with Non Coded Allergen
+	 */
+	@Test
+	public void shouldSaveAllergyWithNonCodedAllergen() throws Exception
+	{
+		Allergy allergy = Context.getPatientService().getAllergyByUuid(getUuid());
+		Patient patient = allergy.getPatient();
+		Allergies allergies = Context.getPatientService().getAllergies(patient);
+		Assert.assertEquals(allergies.size(), 4);
+		
+		// save allergy with non coded allergen
+		String json = "{" +
+				        " \"comment\" : \"allergy comment\"," +
+				        " \"severity\" : { \"uuid\" : \"35d3346a-6769-4d52-823f-b4b234bac3e3\"}," +
+				        " \"allergen\" : " +
+				          " { \"allergenType\" : \"DRUG\", " +
+				          " \"codedAllergen\" : { \"uuid\" : \"5622AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA\"}, " +
+				          " \"nonCodedAllergen\" : \"test non coded allergen\"} " +
+		              "}";
+		
+		// save allergy 
+		SimpleObject savedAllergy = deserialize(handle(newPostRequest(getURI(), json)));
+
+		Assert.assertEquals("allergy comment", Util.getByPath(savedAllergy, "comment"));
+		Assert.assertEquals("35d3346a-6769-4d52-823f-b4b234bac3e3", Util.getByPath(savedAllergy, "severity/uuid"));
+
+		Assert.assertEquals("5622AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA", Util.getByPath(savedAllergy, "allergen/codedAllergen/uuid"));
+		Assert.assertEquals("test non coded allergen", Util.getByPath(savedAllergy, "allergen/nonCodedAllergen"));
+		Assert.assertEquals("DRUG", Util.getByPath(savedAllergy, "allergen/allergenType"));
+
+		// assert that a new allergy has been added
+		allergies = Context.getPatientService().getAllergies(patient);
+		Assert.assertEquals(allergies.size(), 5);
+	}
+	
+	/**
+	 *  Save Allergy with Allergy Reactions
+	 */
+	@Test
+	public void shouldSaveAllergyWithReactions() throws Exception
+	{
+		Allergy allergy = Context.getPatientService().getAllergyByUuid(getUuid());
+		Patient patient = allergy.getPatient();
+		Allergies allergies = Context.getPatientService().getAllergies(patient);
+		Assert.assertEquals(allergies.size(), 4);
+		
+		// update allergy with reactions
+		String json = "{ \"reactions\" : " +
+			       "[" +
+			         "{" +
+					   " \"allergy\" : { \"uuid\" : \"" + allergy.getUuid() + "\"}," +
+			           " \"reaction\" : { \"uuid\" : \"35d3346a-6769-4d52-823f-b4b234bac3e3\" }" +
+			         "}," +
+			         "{" +
+					   " \"allergy\" : { \"uuid\" : \"" + allergy.getUuid() + "\"}," +
+			           " \"reaction\" : { \"uuid\" : \"5622AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA\" }," +
+			           " \"reactionNonCoded\" : \"test non coded reaction\"" +
+			         "}" +
+			       "]" +
+		       "}";
+		SimpleObject savedAllergy = deserialize(handle(newPostRequest(getURI() + "/" + allergy.getUuid(), json)));
+		
+		Assert.assertEquals("35d3346a-6769-4d52-823f-b4b234bac3e3", Util.getByPath(savedAllergy, "reactions[0]/reaction/uuid"));

```

Comment:
The problem is how do I know which allergy in the fetched list was just saved.

Get Outlook for mobile




On Sat, Apr 9, 2016 at 12:36 PM -0700, "dkayiwa" <notifications@github.com> wrote:





> +             String json = "{ \"reactions\" : " +
> +                            "[" +
> +                              "{" +
> +                                        " \"allergy\" : { \"uuid\" : \"" + allergy.getUuid() + "\"}," +
> +                                " \"reaction\" : { \"uuid\" : \"35d3346a-6769-4d52-823f-b4b234bac3e3\" }" +
> +                              "}," +
> +                              "{" +
> +                                        " \"allergy\" : { \"uuid\" : \"" + allergy.getUuid() + "\"}," +
> +                                " \"reaction\" : { \"uuid\" : \"5622AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA\" }," +
> +                                " \"reactionNonCoded\" : \"test non coded reaction\"" +
> +                              "}" +
> +                            "]" +
> +                    "}";
> +             SimpleObject savedAllergy = deserialize(handle(newPostRequest(getURI() + "/" + allergy.getUuid(), json)));
> +
> +             Assert.assertEquals("35d3346a-6769-4d52-823f-b4b234bac3e3", Util.getByPath(savedAllergy, "reactions[0]/reaction/uuid"));

How about here? Can't we just fetch allergies after posting and assert on allergy properties?

---
You are receiving this because you were mentioned.
Reply to this email directly or view it on GitHub:
https://github.com/openmrs/openmrs-module-webservices.rest/pull/195/files/88326ff12836bf08713921e6c12342f22e597ffb#r59120220